jsurfer-core/src/main/java/org/jsfr/json/filter/MatchRegexPredicate.java

```diff
@@ -46,10 +46,16 @@ public MatchRegexPredicate(JsonPath relativePath, Pattern regex) {
     public boolean apply(JsonPath jsonPosition, PrimitiveHolder primitiveHolder, JsonProvider jsonProvider) {
         if (primitiveHolder != null && this.getRelativePath().matchFilterPath(jsonPosition)) {
             Object candidate = primitiveHolder.getValue();
-            String string = (String) jsonProvider.cast(candidate, String.class);
-            return regex.matcher(string).find();
+            return regex.matcher(cast(candidate, jsonProvider)).find();
         }
         return false;
     }
 
+    private static String cast(Object candidate, JsonProvider jsonProvider) {
+        if (candidate instanceof Number || candidate instanceof String) {
```

Comment:
Why wasn't `cast(candidate, jsonProvider)` sufficient?

Reply:
Tried it and saw which tests fail. But I'd say the code should remain here. The issue is in the provider's `cast` method.

Reply:
Yes, providers should have handled that but some, including the one we use in `hazelcast-sql` `JacksonJrProvider`, don't, so we have to work around it for basic types that we know how to convert to String.

Reply:
Boolean also should be there

Reply:
But shouldn't we fix the providers?

Reply:
You're right. For some reason, I thought they are external dependency for us. Will do it.